tests/terratest/functions/set_modules/setEC2RKE1.go

```diff
@@ -0,0 +1,43 @@
+package functions
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/rancher/rancher/tests/terratest/components"
+	"github.com/rancher/rancher/tests/terratest/models"
+)
+
+func SetEC2RKE1(module string, k8sVersion string, nodePools []models.Nodepool, config string, file *os.File) (done bool, err error) {
+	config = config + components.RKE1ClusterPrefix + k8sVersion + components.RKE1ClusterSuffix + components.RKE1EC2NodeTemplate
+	poolConfig := ``
+	clusterSyncNodePoolIDs := `  node_pool_ids = [`
+	num := 1
+	for _, pool := range nodePools {
+		poolNum := strconv.Itoa(num)
+		quantity := strconv.Itoa(pool.Quantity)
+		if pool.Etcd != "true" && pool.Cp != "true" && pool.Wkr != "true" {
+			return false, fmt.Errorf(`no roles selected for pool` + poolNum + `; at least one role is required`)
+		}
+		if pool.Quantity <= 0 {
+			return false, fmt.Errorf(`invalid quantity specified for pool` + poolNum + `; quantity must be greater than 0`)
+		}
+		poolConfig = poolConfig + components.RKE1NodePoolPrefix + poolNum + components.RKE1NodePoolSpecs1 + poolNum + components.RKE1NodePoolSpecs2 + poolNum + `-"` + components.RKE1NodePoolSpecs25 + quantity + components.RKE1NodePoolSpecs3 + pool.Cp + components.RKE1NodePoolSpecs4 + pool.Etcd + components.RKE1NodePoolSpecs5 + pool.Wkr + components.RKE1NodePoolSuffix
```

Comment:
(see question about autogenerated files for more context) - The naming of these is not intuitive for me. I would expect some comment about what these are so that another user can come back and understand what they do, as we will need to maintain this part of the code. What is `RKE1NodePoolSpecs25` vs `RKE1NodePoolSpecs2`? even if I go look at those files, I get
```
var RKE1NodePoolSpecs25 = `
  node_template_id = rancher2_node_template.rancher2_node_template.id
  quantity         = ` 
```
and 
```
var RKE1NodePoolSpecs2 = `"
  hostname_prefix  = "${var.hostname_prefix}-tf-pool`
```
or in other words, Whoever has to go back and maintain this shouldn't have to do so much tracking to figure out what the variable does. Is there a way to name these in a more friendly way, or if not at least comment what this is doing?

Reply:
Id love to discuss this further during PR sync up.  These are essentially building out the main.tf files.  These won't need to be maintained unless the rancher2 provider resource syntax changes.  However, with the growth of this repo, it would be possible to add new fields or values to these blocks, so implementing a better system around this would be ideal.